src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -89,20 +90,100 @@
     private static final long MB = 1024L * 1024;
     private static final long MINIMUM_FREE_SPACE = 200 * MB;
 
+    private static final class DownloadTask extends Task<Void> implements ProgressListener {
```

Comment:
Definitely needs Javadoc, and should probably live in its own class.

Reply:
+1 for Javadoc.. But won't this task be better encapsulated as an inner class, if it's not supposed to be used outside of `ApplicationController`?